src/librustc_middle/hir/mod.rs

```diff
@@ -81,3 +82,28 @@ pub fn provide(providers: &mut Providers<'_>) {
     providers.hir_owner_nodes = |tcx, id| tcx.index_hir(LOCAL_CRATE).map[id].with_bodies.as_deref();
     map::provide(providers);
 }
+
+pub trait TargetExt {
```

Comment:
Can this extension trait go back in librustc_passes? It does not really need to be in librustc_middle.

Reply:
@cjgillot: Ideally, this would be in `rustc_hir` along with the other inherent methods on `Target`. I wanted to place this as close as possible to `Target` in the crate graph, so that it doesn't need to be moved later if it needs to be used in a dependency of `librustc_passes`

Reply:
I think this should either stay the free function it was or become an inherent method.